baseapp/abci.go

```diff
@@ -247,6 +247,10 @@ func (app *BaseApp) DeliverTx(ctx sdk.Context, req abci.RequestDeliverTx) (res a
 	gInfo, result, anteEvents, _, err := app.runTx(ctx.WithTxBytes(req.Tx).WithVoteInfos(app.voteInfos), runTxModeDeliver, req.Tx)
 	if err != nil {
 		resultStr = "failed"
+		// if we have a result, use those events instead of just the anteEvents
+		if result != nil {
+			return sdkerrors.ResponseDeliverTxWithEvents(err, gInfo.GasWanted, gInfo.GasUsed, sdk.MarkEventsToIndex(result.Events, app.indexEvents), app.trace)
+		}
```

Comment:
Are we throwing away the ante events here?

Reply:
no, because the ante events are prepended onto the result events now if the result is non-nil (https://github.com/sei-protocol/sei-cosmos/pull/107/files/af11b782ed830189a7d07e944a69804857d06add#diff-328a55b5ac5816102544792c5c6a75df50df3b9b01afe770c9abc2da79b3e6afR860-R862)